### PR TITLE
Rewritten remember entity shard bugfixes

### PR DESCRIPTION
**Pull request by johanandren**
Fixes one issue where StartEntity and StartEntityAck were not handled correctly in shard and one where the test config had not been updated for separate stores and was trying to use state store mode persistence + ddata remember entities (we have config protection against that but test was setting up internals of sharding itself)

### CODE REVIEW

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala [hunk 1]
```diff
@@ -806,12 +768,37 @@ private[akka] class Shard(
     }
   }
 
+  // FIXME in what scenario do we get this to the shard?
```
Comment (johanandren) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala:771]: I haven't quite figured this out. When remember entities, there is a `StartEntity` -> `StartEntityAck` exchange with the `RememberEntitiesStarter`, and the `StartEntity` can come from the outside, but I don't find anywhere that the shard itself sent the `StartEntity` and should get the ack.

Comment (chbatey) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala:771]: the starter forwards the ack to the shard

Comment (johanandren) [reply akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala:771]: Yep, I figured that out and messed with it in #29164

**Comment by johanandren** [comment]
(merged this so I can continue on other things with the tests passing)

**Pull request merged**